internal/rbd/rbd_util.go

```diff
@@ -1140,6 +1140,32 @@ func updateSnapshotDetails(ctx context.Context, rbdSnap *rbdSnapshot) error {
 	}
 	rbdSnap.VolSize = vol.VolSize
 
+	// get the parent to set rbdSnap.SourceVolumeID
+	parent, err := rbdSnap.getParent()
```

Comment:
we can store the sourceVolumeID in the omap so that we can have 2 advantages
* when the parent is not present we still be able to set the SourceVolumeID which is expected
* SourceVolumeId is a required field in the SPEC that need to be always set
* If source is in different clusterID or something we dont need to worry about it as we dont need to check it in the backend(this might be extra).

Reply:
yeah, I wondered why the SourceVolumeId was not stored in the journal too.